### PR TITLE
upgrades and fixes

### PR DESCRIPTION
**Pull request by rtk-rnjn**
Rewrite bot.py

### CODE REVIEW

**Comment by rtk-rnjn** [comment]
Don't `change_presence` (or make API calls) in `on_ready` within your Bot or Client.
Discord has a high chance to completely disconnect you during the READY or GUILD_CREATE events (1006 close code) and there is nothing you can do to prevent it.

Instead set the `activity` and `status` kwargs in the constructor of these Classes.

```py
bot = commands.Bot(command_prefix="!", activity=..., status=...)
```
As noted in the docs, `on_ready` is also triggered _multiple_ times, not just once. 

Basically: don't 👏 do 👏 shit 👏 in 👏 on_ready.

**Comment by rtk-rnjn** [comment]
**Use aiohttp.**
`requests` and `urllib` are blocking. Do **not** use these libraries within your asynchronous code. 
(http://discordpy.readthedocs.io/en/latest/faq.html#what-does-blocking-mean)

discord.py uses `aiohttp`, so it should already be installed. An example of code using aiohttp and discord.py:

```py
async with aiohttp.ClientSession() as cs:
    async with cs.get('https://httpbin.org/json') as r:
        res = await r.json()  # returns dict
        await ctx.send(res['slideshow']['author'])
```

For more help, see aiohttp's documentation: http://aiohttp.readthedocs.io/en/stable/
